core/codenvy-hosted-sso-server-organization/src/main/java/com/codenvy/auth/sso/server/OrgServiceUserCreator.java

```diff
@@ -81,19 +82,18 @@ public User createUser(String email, String userName, String firstName, String l
             String password = UUID.randomUUID().toString().replace("-", "").substring(0, 12);
 
             try {
-                final User user = new User().withId(id)
-                                            .withName(userName)
-                                            .withEmail(email)
-                                            .withPassword(password);
-                userDao.create(user);
+
+                while (!createNonReservedUser(id, userName, email, password)) {
```

Comment:
on a side note, maybe we should limit the number of "retries"
if for example the server is throwing exception at each call, we will enter into an infinite loop